src/java/org/apache/cassandra/db/Directories.java

```diff
@@ -485,18 +485,39 @@ public boolean hasAvailableDiskSpace(long estimatedSSTables, long expectedTotalW
     {
         long writeSize = expectedTotalWriteSize / estimatedSSTables;
         long totalAvailable = 0L;
+        boolean hasSpace = true;
 
         for (DataDirectory dataDir : paths)
         {
             if (DisallowedDirectories.isUnwritable(getLocationForDisk(dataDir)))
                   continue;
             DataDirectoryCandidate candidate = new DataDirectoryCandidate(dataDir);
             // exclude directory if its total writeSize does not fit to data directory
+            logger.debug("DataDirectory {} has {} bytes available, checking if we can write {} bytes", dataDir.location, candidate.availableSpace, writeSize);
```

Comment:
I don't think we need this , as it may print the log every compaction time not matter the data dir's availableSpace is  really available.
we can remove this.

Reply:
good catch. agree.

Reply:
yeah，I have checked the trunk branch before left the comment and I also saw this debug log message, but I think we should  also remove the debug log there.

Besides I think "logger.debug("FileStore {} has {} bytes available, checking if we can write {} bytes", toWrite.getKey(), availableForCompaction, toWrite.getValue());"   this message just tell us we need a bytes and we have b bytes, if the dir have enough space , the message is meaningless and useless to me. if we do not have enough space, the warn log below this log can give enough information.

Reply:
I was thinking about it when I worked on trunk. I think the message makes sense as such: It records exactly  the kind of environmental data you would use for DEBUG purposes, but 99% of the time is uninteresting. The fact that the result of the function can be determined from the message is actually a good property. It just means a single DEBUG message has captured everything.

Now... I do agree the DEBUG message is perhaps unnecessary after the addition of the WARN message. 

The reason I didn't remove it was that it would have felt weird to remove something that wasn't even released yet. But of course we can do it, it's a poor reason.

But... Maybe ask the authors of the patch that added the debug message @krummas and @jmckenzie-dev ?

Reply:
yeah ，we may ask them for help